Content.Client/Mapping/MappingState.cs

```diff
@@ -0,0 +1,860 @@
+            .Bind(ContentKeyFunctions.MappingPick, new PointerInputCmdHandler(HandlePick, outsidePrediction: true))
+            .Bind(ContentKeyFunctions.MappingRemoveDecal, new PointerInputCmdHandler(HandleEditorCancelPlace, outsidePrediction: true))
+            .Bind(ContentKeyFunctions.MappingCancelEraseDecal, new PointerInputCmdHandler(HandleCancelEraseDecal, outsidePrediction: true))
+            .Bind(ContentKeyFunctions.MappingOpenContextMenu, new PointerInputCmdHandler(HandleOpenContextMenu, outsidePrediction: true))
+            .Register<MappingState>();
+
+        _overlays.AddOverlay(new MappingOverlay(this));
+
+        Screen.Prototypes.UpdateVisible(_prototypes);
+    }
+
+    private bool HandleOpenContextMenu(in PointerInputCmdArgs args)
+    {
+        var coords = args.Coordinates.ToMap(_entityManager, _transform);
+        if (_verbs.TryGetEntityMenuEntities(coords, out var entities))
+            _entityMenuController.OpenRootMenu(entities);
+
+        return true;
+    }
+
+    protected override void Shutdown()
+    {
+        CommandBinds.Unregister<MappingState>();
+
+        Screen.Prototypes.SearchBar.OnTextChanged -= OnSearch;
+        Screen.Prototypes.CollapseAllButton.OnPressed -= OnCollapseAll;
+        Screen.Prototypes.ClearSearchButton.OnPressed -= OnClearSearch;
+        Screen.Prototypes.GetPrototypeData -= OnGetData;
+        Screen.Prototypes.SelectionChanged -= OnSelected;
+        Screen.Prototypes.CollapseToggled -= OnCollapseToggled;
+        Screen.Pick.OnPressed -= OnPickPressed;
+        Screen.Delete.OnPressed -= OnDeletePressed;
+        Screen.EntityReplaceButton.OnToggled -= OnEntityReplacePressed;
+        Screen.EntityPlacementMode.OnItemSelected -= OnEntityPlacementSelected;
+        Screen.EraseEntityButton.OnToggled -= OnEraseEntityPressed;
+        Screen.EraseDecalButton.OnToggled -= OnEraseDecalPressed;
+        _placement.PlacementChanged -= OnPlacementChanged;
+
+        UserInterfaceManager.ClearWindows();
+        _loadController.UnloadScreen();
+        UserInterfaceManager.UnloadScreen();
+        _input.Contexts.GetContext("common").RemoveFunction(ContentKeyFunctions.MappingUnselect);
+        _input.Contexts.GetContext("common").RemoveFunction(ContentKeyFunctions.SaveMap);
+        _input.Contexts.GetContext("common").RemoveFunction(ContentKeyFunctions.MappingEnablePick);
+        _input.Contexts.GetContext("common").RemoveFunction(ContentKeyFunctions.MappingEnableDelete);
+        _input.Contexts.GetContext("common").RemoveFunction(ContentKeyFunctions.MappingPick);
+        _input.Contexts.GetContext("common").RemoveFunction(ContentKeyFunctions.MappingRemoveDecal);
+        _input.Contexts.GetContext("common").RemoveFunction(ContentKeyFunctions.MappingCancelEraseDecal);
+        _input.Contexts.GetContext("common").RemoveFunction(ContentKeyFunctions.MappingOpenContextMenu);
```

Comment:
Should this just have its own input context that inherits common instead?

Reply:
It would be possible but then you would need to change the context when entering/exiting the mapping state.

Reply:
Does that affect aghost or anything.

Reply:
I don't think so, but it would conflict with the "editor" context of the F5/F6 windows.

Reply:
This code makes me weep but I'll put it on the pile of things to fix when we nuke placement manager.